src/handlers/testing/align/components/App.tsx

```diff
@@ -0,0 +1,218 @@
+import { Box, Text, render, useApp } from 'ink';
+import { useEffect, useState } from 'react';
+import type { AsyncHandlerOf, Handler } from '../../../../util/types';
+import { EventName, emitter } from '../util/emitter';
+import {
+  AfterGradeAction,
+  type AutogeneratedEvaluator,
+  type TestCaseResult,
+  type GradedTestCaseResult,
+} from '../util/models';
+import { AutogeneratedEvaluatorsList } from './AutogeneratedEvaluatorsList';
+import { GradeTestCaseResultForm } from './GradeTestCaseResultForm';
+import { GradedTestCaseResultsList } from './GradedTestCaseResultsList';
+import { WaitingFor } from './WaitingFor';
+
+interface AppProps {
+  testExternalId: string;
+  sessionId: string;
+  testCaseHashes: () => string[];
```

Comment:
Why does this need to be an arrow function?

Reply:
in theory the hashes could change between runs if the test suite was edited but i can just assume that won't happen for now